src/braket/experimental/autoqasm/program/program.py

```diff
@@ -171,52 +194,40 @@ def get_oqpy_program(self) -> oqpy.Program:
         """
         return self.oqpy_program_stack[-1]
 
-    class OqpyProgramContextManager:
-        """Context manager responsible for managing the oqpy programs which are used
-        by the ProgramConversionContext."""
-
-        def __init__(self, oqpy_program: oqpy.Program, oqpy_program_stack: List[oqpy.Program]):
-            self.oqpy_program = oqpy_program
-            self.oqpy_program_stack = oqpy_program_stack
-
-        def __enter__(self):
-            self.oqpy_program_stack.append(self.oqpy_program)
-
-        def __exit__(self, exc_type, exc_value, exc_tb):
-            self.oqpy_program_stack.pop()
-
-    def push_oqpy_program(self, oqpy_program: oqpy.Program) -> OqpyProgramContextManager:
+    @contextlib.contextmanager
```

Comment:
For consistency with the new code, I also refactored/simplified the two existing context managers here to use `@contextlib.contextmanager` rather than manually implementing a class with `__enter__` and `__exit__`.

Reply:
Great improvement!